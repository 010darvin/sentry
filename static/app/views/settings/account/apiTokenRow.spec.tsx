import {ApiToken} from 'fixtures/js-stubs/apiToken.js';

import {render, screen, userEvent} from 'sentry-test/reactTestingLibrary';

import ApiTokenRow from 'sentry/views/settings/account/apiTokenRow';

describe('ApiTokenRow', () => {
  it('renders', () => {
    const wrapper = render(<ApiTokenRow onRemove={() => {}} token={ApiToken()} />);

    // Should be loading
    expect(wrapper.container).toSnapshot();
  });

  it('calls onRemove callback when trash can is clicked', () => {
    const cb = jest.fn();
    render(<ApiTokenRow onRemove={cb} token={ApiToken()} />);

    userEvent.click(screen.getByRole('button'));
    expect(cb).toHaveBeenCalled();
  });
});
