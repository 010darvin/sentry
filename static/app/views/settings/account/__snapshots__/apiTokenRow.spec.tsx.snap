// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApiTokenRow renders 1`] = `
<div>
  <div
    class="css-s6dcze-PanelItem-StyledPanelItem eu9lu7u7"
  >
    <div
      class="css-1yb85im-Controls eu9lu7u6"
    >
      <div
        class="css-fgwtz1-InputWrapper eu9lu7u5"
      >
        <div
          class="css-1biel1q-InputGroupWrap e13j5zq05"
        >
          <input
            class="eqx5lw01 css-182ojs2-Input-inputStyles-StyledInput-getInputStyles-StyledInput e13j5zq03"
            readonly=""
            value="apitoken123"
          />
          <div
            class="css-vdn3ll-InputItemsWrap-InputTrailingItemsWrap e13j5zq00"
            data-test-id="input-trailing-items"
          />
        </div>
      </div>
      <button
        aria-disabled="false"
        aria-label="Remove"
        class="css-c98s0k-StyledButton-getColors-getButtonStyles e1b8u3ky2"
        role="button"
        type="button"
      >
        <span
          class="css-vul34l-InteractionStateLayer-InteractionStateLayer-InteractionStateLayer e1fwp9rx0"
          role="presentation"
        />
        <span
          class="css-1z0n9wg-ButtonLabel e1b8u3ky1"
        >
          <span
            class="css-1ywt59b-Icon e1b8u3ky0"
          >
            <svg
              data-test-id="icon-subtract"
              fill="currentColor"
              height="12px"
              viewBox="0 0 16 16"
              width="12px"
            >
              <path
                d="M8,16a8,8,0,1,1,8-8A8,8,0,0,1,8,16ZM8,1.53A6.47,6.47,0,1,0,14.47,8,6.47,6.47,0,0,0,8,1.53Z"
              />
              <path
                d="M11.28,8.75H4.72a.75.75,0,1,1,0-1.5h6.56a.75.75,0,1,1,0,1.5Z"
              />
            </svg>
          </span>
          Remove
        </span>
      </button>
    </div>
    <div
      class="css-7fc9vz-Details eu9lu7u4"
    >
      <div
        class="css-7ovylo-ScopesWrapper eu9lu7u3"
      >
        <div
          class="css-1g8z09y-Heading eu9lu7u0"
        >
          Scopes
        </div>
        <div
          class="css-avjeiu-ScopeList eu9lu7u2"
        >
          scope1, scope2
        </div>
      </div>
      <div>
        <div
          class="css-1g8z09y-Heading eu9lu7u0"
        >
          Created
        </div>
        <time
          class="css-szkkrh-Time eu9lu7u1"
        >
          <time>
            Jan 12, 2018 2:01 AM
          </time>
        </time>
      </div>
    </div>
  </div>
</div>
`;
